pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/tasks/PipInstallTask.groovy

```diff
@@ -87,87 +87,110 @@ class PipInstallTask extends DefaultTask implements FailureReasonProvider {
     }
 
     /**
+     * Method that checks to ensure that the current project is prepared to pip install.  It ignores the
+     * base pygradle libraries
+     *
+     * @param installable file object for the system to install
+     * @return
+     */
+    private boolean isReadyForInstall(File installable) {
+        if (installable.absolutePath == project.projectDir.absolutePath) {
+            /*
+            we are installing the product itself.  Now lets see if its ready for it
+            this ignores dependencies
+             */
+            def setupPyFile = new File(installable.absolutePath, "setup.py")
+            if (!setupPyFile.exists()) {
+                logger.lifecycle(PythonHelpers.createPrettyLine("Install ${project.name}", "[ABORTED]"))
+                project.logger.warn("setup.py missing, skipping venv install for product ${project.name}.  Run 'gradle generateSetupPy' to generate a generic file")
+                return false
+            }
+        }
+        return true
+    }
+
+    /**
      * Install things into a virtual environment.
      * <p>
      * Always run <code>pip install --no-deps <args></code> with arguments. This prevents
      * naughty dependencies from using pure-Setuptools and mppy machinery to
      * express and download dependencies.
      */
     @TaskAction
-    public void pipInstall() {
-
+    void pipInstall() {
         def pyVersion = pythonDetails.getPythonVersion().pythonMajorMinor
         def extension = ExtensionUtils.getPythonExtension(project)
         def sitePackages = findSitePackages()
 
         for (File installable : getConfigurationFiles()) {
+            if (isReadyForInstall(installable)) {
```

Comment:
we don't use this plugin for python projects (yet, thats phase 2)  phase 1 is to enable sphinx documentation for our java, c#, Javascript, and sql projects, several hundred of them.  So for a java project that has nothing to do with python, we can't have the plugin fail because it couldn't find a python project.  This method is looking to see if pip is getting ready to install the python project itself.  If it is, is it properly configured (setup.py exists and such)?  If its not properly configured, we throw a warning and move on instead of blow up the build.